src/csharp/Microsoft.Spark/Sql/PicklingUdfWrapper.cs

```diff
@@ -4,9 +4,16 @@
 
 using System;
 using Microsoft.Spark.Utils;
+using static Microsoft.Spark.Sql.PicklingUdfWrapperHelper;
 
 namespace Microsoft.Spark.Sql
 {
+    internal static class PicklingUdfWrapperHelper
+    {
+        internal static T CastOrConvertTo<T>(ref bool? same, object param) =>
+            (same ??= param is T) ? (T)param : TypeConverter.ConvertTo<T>(param);
```

Comment:
Would this be inlined? The whole purpose of this PR was perf, so can you check if this wouldn't cause regression?

Reply:
Removed fn call.